File: apps/expo/modules/bible-strong-app-check/src/BibleStrongAppCheckModule.ts
import { NativeModule, requireOptionalNativeModule } from 'expo-modules-core'

declare class BibleStrongAppCheckModule extends NativeModule {
  readonly provider: 'playIntegrity' | 'recaptchaEnterprise'
  initialize(appName: string): Promise<void>
  getToken(forceRefresh: boolean): Promise<{ token: string }>
}

// Older Android binaries and Apple builds continue to use the RNFirebase providers.
export default requireOptionalNativeModule<BibleStrongAppCheckModule>('BibleStrongAppCheck')

File: apps/expo/modules/bible-strong-app-check/android/src/standard/java/expo/modules/biblestrongappcheck/BibleStrongAppCheckModule.kt
package expo.modules.biblestrongappcheck

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class BibleStrongAppCheckModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("BibleStrongAppCheck")
    Constant("provider") { "playIntegrity" }
  }
}
